### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by SaboorSohaib**
# Hi Dear :wave: 

**In this milestone, I implemented the following requirements**

- [ ] Add the following validations:
 - For the User model:
   - Name must not be blank.
   - PostsCounter must be an integer greater than or equal to zero.
 - For the Post model:
   - Title must not be blank.
   - Title must not exceed 250 characters.
   - CommentsCounter must be an integer greater than or equal to zero.
   - LikesCounter must be an integer greater than or equal to zero.
 - Add unit specs for all of your models' methods and validations.

**Thank you**

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    expect(subject).to_not be_valid
+  end
+
+  it ' Should test posts_counter to be >= 0' do
+    expect(subject.posts_counter).to be >= 0
+  end
+
+  it 'can display most recent posts' do
+    expect(subject).to respond_to(:most_recent_posts)
+  end
```
Comment (julie-ify) [inline spec/models/user_spec.rb]: - [ ] Kindly add tests that show the most recent posts of the user are actually returned. You may need to create some posts in this file

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    expect(subject.likes_counter).to be >= 0
+  end
+
+  it 'can update posts counter' do
+    expect(subject).to respond_to(:update_posts_counter)
+  end
+
+  it 'can display most recent comments' do
+    expect(subject).to respond_to(:most_recent_comments)
+  end
```
Comment (julie-ify) [inline spec/models/post_spec.rb]: - [ ] Good job adding tests that show the post object responds to the methods. However, you need to add more comprehensive tests that demonstrate the methods behave as expected. Kindly add tests (or modify these ones):

  - that shows that the `update_posts_counter` actually updates the posts counter of the post author.
  - that the `most_recent_comments` method fetches the recent comments of the post object. You may need to create comments in this file and ensure the expected comments match the ones you created.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @SaboorSohaib,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 2 above)
Comment (SaboorSohaib) [reply spec/models/post_spec.rb]: ## Hi @julie-ify 

Thank you very much for your review. Please can you give me a hint to solve this challenge?

**Review by DammyShittu — APPROVED** [review]
Hi @SaboorSohaib,

## APPROVED 🟢

## Highlights
✔️ Tests cover model custom methods

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**